### ISSUE TITLE
App default nonce for user includes nonces of 'expired' Txs; Txs won't work

### ISSUE THREAD

**Issue by tomstuart123** [decentdao/decent-interface#1994]
During testing of 1990, I was using [this Safe](https://deploy-preview-1990.app.dev.decentdao.org/home?dao=sep:0x773eC5B85aebAa323e700E9bbAA7Cb2eFeAdf56F)

As you can see the prior Txs were:
- Safe name - default to nonce 1 (executed)
- Add ChildSafe - default to nonce 2 (rejected)
- Update safename - default to nonce 3 (blocked by 2 above)
- Add signer (I. manually had to set it to nonce2, not 4)

The Problem - The 'Add ChildSafe' proposal that at nonce 2 ran out of time and was never executed. However, our nonce assumptions still allowed for it and incremented the default nonce to 3 after the expiration time. This meant my 'update safename' was blocked. As SafeName has no manual nonce, I couldn't even get by it (until I used the 'Add Signer' proposal to use the correct nonc2 and unblock it).

Thinking - the app was correct originally to default to nonce 3 as the childsafe proposal was in 'pending'/. However, once it expires, pending should be no longer accounted for. Our app didn't do this

The solution:
- check our nonce locations and make sure they default to the correct nonce if a pending proposal has now expired. note - I know for sure, addSigner and safeName edits don't do this right now. Not sure about the others

Stretch goal:
- eventually we should probably allow manual nonce for all proposals  (the missing ones are safename and snapshot space). However, this is debateable so no need to tackle this in this issue

**Comment by adamgall** [comment]
I think I understand what's going on here. This is indeed a bug that should be fixed.

**Comment by Da-Colon** [comment]
Is this still an issue? haven't let an proposal expire in a while
